client/consul_template_test.go

```diff
@@ -905,3 +911,125 @@ func TestTaskTemplateManager_Signal_Error(t *testing.T) {
 		t.Fatalf("Unexpected error: %v", harness.mockHooks.KillReason)
 	}
 }
+
+// TestTaskTemplateManager_Env asserts templates with the env flag set are read
+// into the task's environment.
+func TestTaskTemplateManager_Env(t *testing.T) {
+	template := &structs.Template{
+		EmbeddedTmpl: `
+# Comment lines are ok
+
+FOO=bar
+foo=123
+ANYTHING-goes=Spaces are=ok!
```

Comment:
are newlines okay too? 

assume key `derp` has a value in Consul of the following, and you do DERP={{ key "derp" }}

```
hello
world
```